client/main/editors/transformeditor.js

```diff
@@ -0,0 +1,492 @@
+        }
+
+        let $fp = $('<div class="formula-list-item"></div>').appendTo($formulaPair);
+
+        elements.$focusedFormula = null;
+        if (elements.$formulas === undefined)
+            elements.$formulas = [ ];
+
+        let $formula = $('<div class="formula" type="text" placeholder="' + _sign + 'e.g. ' + _example + '" contenteditable="true" data-index="' + index + '" style="text-indent:' + indent + '">' + formula + '</div>').appendTo($fp);
+        $('<div class="equal">' + prefix + '</div>').appendTo($fp);
+
+        elements.$formulas.push($formula);
+
+        let $opEdit = null;
+
+        $formula.on('blur', (event) => {
+            this.formula[($formulaBox.index() * 2) + index] = $formula[0].textContent;
+            if (hasOp && elements._editorClicked === false)
+                $opEdit.hide();
+        });
+
+        $formula.on('focus', (event) => {
+            elements.$focusedFormula = $formula;
+            this._focusFormulaControls();
+            if (this.formulasetup.focusedOn() !== $formula)
+                this.formulasetup.show($formula, '', true);
+            if (hasOp)
+                $opEdit.show();
+        });
+
+        $formula.on('input', (event) => {
+            dropdown.updatePosition();
+
+            if (hasOp) {
+                let count = this._startsWithValidOps($formula);
+                if (count !== 0)
+                    $opEdit.css('width', (count+1) + 'ch');
+
+                if (dropdown.content() === this.opsToolbar)
+                    dropdown.hide();
+            }
+        });
+
+        if (hasOp) {
+            $opEdit = $('<div class="down-arrow">a</div>').appendTo($fp);
+            $opEdit.css('width', _sign.length + 'ch');
+            $opEdit.hide();
+
+            $opEdit.on('click', (event) => {
+                if (this._$wasEditingOpsFormula !== $formula || dropdown.content !== this.opsToolbar) {
+                    this.opsToolbar.show($formula);
+                    dropdown.show($formula, this.opsToolbar);
+
+                    let sel = window.getSelection();
+                    /*let range = sel.getRangeAt(0);
+                    let start = range.startOffset;
+                    let end = range.endOffset;
+                    let text = el.textContent;
+                    let before = text.substring(0, start);
+                    let after  = text.substring(end, text.length);*/
+
+                    let count = this._startsWithValidOps($formula);
+
+                    /*let textSelected = text.substring(start, end);
+                    el.textContent = (before + newText + after);*/
+                    sel.setBaseAndExtent($formula[0].firstChild, 0, $formula[0].firstChild, count);
+                    $formula[0].focus();
+                    //elements.$formula.focus();
+                    $opEdit.addClass('is-active');
+                }
```

Comment:
did you want to tidy this block up? or was this intentional?

Reply:
this is file is not fully finished and the commented out code may still be useful to me. So its untidy but part of a work in progress. i can finish the file off but it'll take a few days before i could commit. Part of the commit the usable code so that others can use while the other polishing occurs.